src/Sql/dbo/Stored Procedures/OrganizationUser_DeleteByIds.sql

```diff
@@ -25,22 +25,74 @@ BEGIN
         EXEC [dbo].[SsoUser_DeleteMany] @UserAndOrganizationIds
     END
 
-    DELETE CU
-    FROM
-        [dbo].[CollectionUser] CU
+    
+    DECLARE @BatchSize INT = 100
+
+-- Delete SSO Users
+WHILE @BatchSize > 0
+    BEGIN
+    BEGIN TRANSACTION SsoUser_DeleteMany_SsoUsers
+
+    DELETE TOP(@BatchSize) SU
+        FROM
+        [dbo].[SsoUser] SU
     INNER JOIN
-        @Ids I ON I.Id = CU.OrganizationUserId
+        #SSOIds ON #SSOIds.Id = SU.Id
 
-    DELETE GU
-    FROM
-        [dbo].[GroupUser] GU
+    SET @BatchSize = @@ROWCOUNT
+
+    COMMIT TRANSACTION SsoUser_DeleteMany_SsoUsers
+END
+
+    -- Delete CollectionUsers
+    WHILE @BatchSize > 0
+    BEGIN
+        BEGIN TRANSACTION CollectionUser_DeleteMany_CollectionUsers
+
+        DELETE TOP(@BatchSize) CU
+        FROM
+            [dbo].[CollectionUser] CU
     INNER JOIN
-        @IDs I ON I.Id = GU.OrganizationUserId
+            @Ids I ON I.Id = CU.OrganizationUserId
```

Comment:
Indented too far.

Reply:
Sorry about all these, my sql formatter is gung-ho and very wrong